File: find_town_judge.py
# TC: O(N)
# sc: O(N)

class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        dict1={}
        if n==1 and len(trust)==0:
            return n
        if n==0 or len(trust)==0:
            return -1
        
        dict2={}
        for i in range(1,n+1):
            dict1[i]=[]
            dict2[i]=0
        
        
        for t in trust:
            if t[0] in dict1:
                dict1[t[0]].append(t[1])
            if t[1] not in dict2:
                dict2[t[1]]=1
            else:
                dict2[t[1]]+=1

        for i in dict1:
            if dict1[i]==[] and dict2[i]==n-1:
                return i
        
        return -1
            
        
            
        

File: FindTownJudge.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Graph
{
    class FindTownJudgeLC
    {
        //TC:O(V+E)
        //SC:O(n)
        public int FindJudge(int n, int[][] trust)
        {
            if (n == 0)
            {
                return -1;
            }
            int[] indegrees = new int[n];
            foreach (int[] t in trust)
            {
                indegrees[t[0] - 1]--;
                indegrees[t[1] - 1]++;
            }
            for (int i = 0; i < n; i++)
            {
                if (indegrees[i] == n - 1)
                {
                    return i + 1;
                }
            }
            return -1;
        }
    }
}
